IdentityCore/src/cache/key/MSIDDefaultCredentialCacheKey.m

```diff
@@ -62,7 +64,12 @@ - (NSString *)serviceWithType:(MSIDCredentialType)type
     {
         service  = [NSString stringWithFormat:@"%@|%@", service, appKey];
     }
-    
+
+    if (![NSString msidIsStringNilOrBlank:requestedClaims])
+    {
+        service  = [NSString stringWithFormat:@"%@|%@", service, requestedClaims.msidTokenHash];
```

Comment:
Once we save AT with claims, we won't be able to query it without claims anymore, right?

Reply:
Correct, by design. 

MSAL CPP will have a slightly different behavior when claims/capabilities are in use. To be specific, if claims are set, unlike other MSALs we do not bypass cache, mirroring the behavior of WAM, where claims are part of the cache key. Therefore, to error on the side of caution, MSAL CPP will be saving AT's with requested_claims set where appropriate, and using the requested_claims key when querying cache. 

I'm prioritizing interoperability between MSAL CPP and Obj-C ADAL/MSAL here, by explicitly not returning an AT w/ claims set when no claims are requested on the cache lookup, and vice versa. This way, existing MSAL's do not have to make any change (just have requested_claims = nil), and MSAL CPP's AT's with claims will coexist side by side, no chance of overwriting or conflicts.